rust/src/ikev2/ikev2.rs

```diff
@@ -0,0 +1,715 @@
+    /// List of transactions for this session
+    transactions: Vec<IKEV2Transaction>,
+
+    /// Events counter
+    events: u16,
+
+    /// tx counter for assigning incrementing id's to tx's
+    tx_id: u64,
+
+    /// The connection state
+    connection_state: IKEV2ConnectionState,
+
+    /// The transforms proposed by the initiator
+    pub client_transforms : Vec<Vec<IkeV2Transform>>,
+
+    /// The transforms selected by the responder
+    pub server_transforms : Vec<Vec<IkeV2Transform>>,
+
+    /// The encryption algorithm selected by the responder
+    pub alg_enc:  IkeTransformEncType,
+    /// The authentication algorithm selected by the responder
+    pub alg_auth: IkeTransformAuthType,
+    /// The PRF algorithm selected by the responder
+    pub alg_prf:  IkeTransformPRFType,
+    /// The Diffie-Hellman algorithm selected by the responder
+    pub alg_dh:   IkeTransformDHType,
+    /// The extended sequence numbers parameter selected by the responder
+    pub alg_esn:  IkeTransformESNType,
+
+    /// The Diffie-Hellman group from the server KE message, if present.
+    pub dh_group: IkeTransformDHType,
+
+}
+
+#[derive(Debug)]
+pub struct IKEV2Transaction {
+    /// The IKEV2 reference ID
+    pub xid: u64,
+
+    pub hdr: IkeV2Header,
+
+    pub payload_types: Vec<IkePayloadType>,
+    pub notify_types: Vec<NotifyType>,
+
+    /// IKEv2 errors seen during exchange
+    pub errors: u32,
+
+    /// The internal transaction id
+    id: u64,
+
+    /// The detection engine state, if present
+    de_state: Option<*mut core::DetectEngineState>,
+
+    /// The events associated with this transaction
+    events: *mut core::AppLayerDecoderEvents,
+
+    logged: applayer::LoggerFlags,
+}
+
+
+
+impl IKEV2State {
+    pub fn new() -> IKEV2State {
+        IKEV2State{
+            transactions: Vec::new(),
+            events: 0,
+            tx_id: 0,
+            connection_state: IKEV2ConnectionState::Init,
+            dh_group: IkeTransformDHType::None,
+            client_transforms: Vec::new(),
+            server_transforms: Vec::new(),
+            alg_enc: IkeTransformEncType::ENCR_NULL,
+            alg_auth: IkeTransformAuthType::NONE,
+            alg_prf: IkeTransformPRFType::PRF_NULL,
+            alg_dh: IkeTransformDHType::None,
+            alg_esn: IkeTransformESNType::NoESN,
+        }
+    }
+}
+
+impl IKEV2State {
+    /// Parse an IKEV2 request message
+    ///
+    /// Returns The number of messages parsed, or -1 on error
+    fn parse(&mut self, i: &[u8], direction: u8) -> i8 {
+        match parse_ikev2_header(i) {
+            IResult::Done(rem,ref hdr) => {
+                SCLogDebug!("parse_ikev2: {:?}",hdr);
+                if rem.len() == 0 && hdr.length == 28 {
+                    return 1;
+                }
+                // Rule 0: check version
+                if hdr.maj_ver != 2 || hdr.min_ver != 0 {
+                    SCLogInfo!("Unknown header version: {}.{}", hdr.maj_ver, hdr.min_ver);
```

Comment:
this should be an event without a SCLogInfo

Reply:
Changed to a `MalformedData` event